src/zenml/zen_server/routers/flavors_endpoints.py

```diff
@@ -102,60 +113,50 @@ def get_flavor(
 @handle_exceptions
 def create_flavor(
     flavor: FlavorRequest,
-    auth_context: AuthContext = Security(
-        authorize, scopes=[PermissionType.WRITE]
-    ),
+    _: AuthContext = Security(authorize),
 ) -> FlavorResponse:
     """Creates a stack component flavor.
 
     Args:
         flavor: Stack component flavor to register.
-        auth_context: Authentication context.
 
     Returns:
         The created stack component flavor.
-
-    Raises:
-        IllegalOperationError: If the workspace or user specified in the stack
-            component flavor does not match the current workspace or authenticated
-            user.
     """
-    if flavor.user != auth_context.user.id:
-        raise IllegalOperationError(
-            "Creating flavors for a user other than yourself "
-            "is not supported."
-        )
-
-    created_flavor = zen_store().create_flavor(
-        flavor=flavor,
+    return verify_permissions_and_create_entity(
```

Comment:
`FlavorRequest` doesn't inherit from `UserScopedRequest`. This means that anyone can theoretically create a flavor belonging to another user. Theoretically, it could inherit from `WorkspaceScopedRequest`, but it can't because the user (owner) can be Null in case of the built-in flavors. This looks a lot like the default stack/components use-case, where you used `server_owned_request_model` and an internal request model, maybe the same can be applied here (i.e. `register_builtin_flavors` could use an internal flavor request model).

Reply:
I think it can't inherit from `WorkspaceScopedRequest`/`WorkspaceScopedResponse` as our builtin/integration flavors are registered without a workspace ID so we don't duplicate them in the DB in case someone creates multiple workspaces.

I like your idea of going with the `server_owned_request_model`, I'll try to inherit from the `UserScoped..` classes and see if that works!